### PR TITLE
[android] Preferences to AndroidX, Gson deprecation warning, Build Tools Update

### PR DESCRIPTION
**Pull request by TacoTheDank**
The last one corrects the xml preferences, as the old way is not correct (as documented by Android's coding docs).

### CODE REVIEW

**Comment by gubatron** [comment]
>For further contributions
>
> Create a branch with a descriptive name of the issue you are solving

please next time submit changes to a feature branch.
Now I'll have to cherry-pick each one of these on a temp branch in order to test.

**Comment by gubatron** [comment]
when you build, don't you get this Warning after bumping the tools?


Daemon will be stopped at the end of the build stopping after processing
WARNING: **The specified Android SDK Build Tools version (28.0.3) is ignored**, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.

**Comment by gubatron** [comment]
Left a comment, can't yet upgrade build tools
https://github.com/frostwire/frostwire/commit/cadfa85dec55440e5abb01b06cc1a1d281ebd8be

**Comment by gubatron** [comment]
These two have been merged
- [x] Fix a Gson deprecation
- [x] Build tools update

File: android/res/xml/settings_application.xml [hunk 1]
```diff
@@ -17,11 +17,11 @@
  */
 -->
 <!-- see ApplicationPreferencesFragment -->
-<androidx.preference.PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
-    <androidx.preference.PreferenceCategory
+<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (gubatron) [inline android/res/xml/settings_application.xml:20]: why move it back to PreferenceScreen? isn't androidx the new way?

Comment (gubatron) [reply android/res/xml/settings_application.xml:20]: AndroidX is a major improvement to the original Android Support Library. Like the Support Library, AndroidX ships separately from the Android OS and provides backwards-compatibility across Android releases. AndroidX fully replaces the Support Library by providing feature parity and new libraries. In addition AndroidX includes the following features:

All packages in AndroidX live in a consistent namespace starting with the string androidx. The Support Library packages have been mapped into corresponding androidx.* packages. For a full mapping of all the old classes and build artifacts to the new ones, see the Package Refactoring page.

Comment (gubatron) [reply android/res/xml/settings_application.xml:20]: We went for the old .v7 PreferenceScreen to the new stable androidx API 14 months ago
https://github.com/frostwire/frostwire/commit/76e522825eda2cab7a16045f484c563b8b552eac#diff-579144e5114ee0bc6d4a0fb854995a8a

Comment (gubatron) [reply android/res/xml/settings_application.xml:20]: I see, now `PreferenceScreen` is implicitly `androidx`'s

Merging. Works all the way down to Android 5.0

**Comment by gubatron** [comment]
Merged by cherry-picking, and put the 2 commits of preferences as a single one.

**Comment by gubatron** [comment]
Thank you for these patches, please send newer patches in separate feature branches, a lot easier to work with on my end when reviewing and testing.

**Comment by TacoTheDank** [comment]
Ah, ok, thank you.

**Pull request closed without merging**